networkd/network_server.ml

```diff
@@ -89,6 +89,22 @@ let set_dns_interface _ dbg ~name =
  * to all incoming packets, which confuses certain guests OSes. The workaround
  * constitutes adding a VLAN0 Linux device to strip those headers again.
  *)
+
+let is_vlan_in_use _ dbg ~vlan =
+	(* Identify the vlan is used by FCoE which is unknown to XAPI *)
+	Debug.with_thread_associated dbg (fun () ->
+		let vlan_dir = "/proc/net/vlan" in
+		let vlans =
+			try
+				let vlan_list = Array.to_list (Sys.readdir vlan_dir) in
+				List.map (fun vlan -> List.hd (String.split '-' vlan)) vlan_list
+			with e ->
+				error "Error: could not read %s." vlan_dir;
+				[]
+		in
+		if List.mem vlan vlans then true else false
```

Comment:
It's enough to say `List.mem vlan vlans` - the result is the boolean you are looking for